### PR TITLE
fix(@clayui/css): Adapt Sheet to fit Commerce patterns

### LINKED ISSUE #3762: Adapting sheet to commerce needs
Commerce has a request to adapt sheet component to their needs. 

Commerce product interface has multiple forms in a single screen which is convenient for their use case and users. These individual forms need of a header that allows quick identification. Lexicon has added a header definition to sheet component.

[LEXI-1032](https://issues.liferay.com/browse/LEXI-1032) collects the Commerce proposal and Lexicon answer, together with the team request validation.

Thanks!

### PR DESCRIPTION
**Pull request by pat270**
We should be able to use `sheet` for these components:
https://liferay.design/lexicon/core-components/forms/forms-sheet/#multiple-form
https://liferay.design/lexicon/core-components/forms/forms-sheet/#dataset-content

fixes #3762

### CODE REVIEW

File: clayui.com/content/docs/components/markup-forms-hierarchy.md [hunk 1]
```diff
@@ -652,3 +656,687 @@ Nest `autofit` utilities with `component-title` in `sheet-subtitle` to verticall
 	</span>
 </h4>
 ```
+
+## Variants(#css-variants)
+
+## Sheet Multiple Form(#css-sheet-multiple-form)
```
Comment (matuzalemsteles) [inline clayui.com/content/docs/components/markup-forms-hierarchy.md:662]: Hey @pat270you could add in the examples the code example about the ` ```html... `

**Review by matuzalemsteles — COMMENTED** [review]
@pat270 from what I understand here we will only have new Sheet variations and some new utils to support, right? I just added a comment before moving on with it.

@bryceosterhaus maybe we should support these new variations in our components? what do you think?

File: clayui.com/content/docs/components/markup-forms-hierarchy.md (hunk 1 above)
Comment (pat270) [reply clayui.com/content/docs/components/markup-forms-hierarchy.md:662]: I'll add it.

**Comment by pat270** [comment]
@matuzalemsteles yeah, I think the only new utils are `autofit-padded-no-gutters-sm` and `autofit-col-shrink`.

**Comment by bryceosterhaus** [comment]
@matuzalemsteles yeah we should support these in our components, I can add that today and will tag onto this issue

**Comment by bryceosterhaus** [comment]
Added api for components at https://github.com/liferay/clay/pull/3886. We can merge that PR once this one is through

**Comment by matuzalemsteles** [comment]
Thanks @bryceosterhaus!

**Comment by pat270** [comment]
@matuzalemsteles added the code examples

**Comment by marcoscv-work** [comment]
LGTM!!!!

**Review by matuzalemsteles — APPROVED** [review]
LGTM!

**Pull request merged**